miden/tests/integration/main.rs

```diff
@@ -77,3 +67,43 @@ macro_rules! build_op_test {
         $crate::build_test!(&source, $($tail)+)
     }};
 }
+
+/// Returns a Test struct from the provided source string, debug mode boolean flag and any specified
+/// stack and advice inputs.
+///
+/// Parameters are expected in the following order:
+/// `source`, `in_debug_mode`, `stack_inputs` (optional), `advice_tape` (optional), `advice_sets` (optional)
+///
+/// * `source`: a well-formed source string.
+/// * `in_debug_mode` : boolean flag indicated whether to build test in debug mode.
+/// * `stack_inputs` (optional): the initial inputs which must be at the top of the stack before
+/// executing the `source`. Stack inputs can be provided independently without any advice inputs.
+/// * `advice_tape` (optional): the initial advice tape values. When provided, `stack_inputs` and
+/// `advice_sets` are also expected.
+/// * `advice_sets` (optional): the initial advice set values. When provided, `stack_inputs` and
+/// `advice_tape` are also expected.
+#[macro_export]
+macro_rules! build_debug_test {
+    ($source:expr, $in_debug_mode:expr) => {{
+        $crate::helpers::Test::new($source, $in_debug_mode)
+    }};
+    ($source:expr, $in_debug_mode:expr, $stack_inputs:expr) => {{
+        let inputs = $crate::helpers::ProgramInputs::new($stack_inputs, &[], vec![]).unwrap();
+
+        $crate::helpers::Test {
+            source: String::from($source),
+            inputs,
+            in_debug_mode: $in_debug_mode,
+        }
+    }};
+    ($source:expr, $in_debug_mode:expr, $stack_inputs:expr, $advice_tape:expr, $advice_sets:expr) => {{
+        let inputs =
+            $crate::helpers::ProgramInputs::new($stack_inputs, $advice_tape, $advice_sets).unwrap();
+
+        $crate::helpers::Test {
+            source: String::from($source),
+            inputs,
+            in_debug_mode: $in_debug_mode,
+        }
+    }};
+}
```

Comment:
This approach works, and I like that you didn't duplicate the functionality between the two macros. However, we can actually go even further by making use of the [`tt` fragment specifier](https://veykril.github.io/tlborm/decl-macros/minutiae/fragment-specifiers.html#tt), which will allow us to use the exact same parameter formats for both `build_test` and `build_debug_test`. 

`tt` lets us match all inputs but defer parsing them so that we can pass them to a different macro where they'll be parsed and handled. This allows the following:

```
#[macro_export]
macro_rules! build_test {
    ($($params:tt)+) => {{
        $crate::build_test_by_mode!(false, $($params)+)
    }}
}

#[macro_export]
macro_rules! build_debug_test {
    ($($params:tt)+) => {{
        $crate::build_test_by_mode!(true, $($params)+)
    }}
}

/// This macro is an internal test builder, and is not intended to be called directly from tests. 
/// Instead, the `build_test` and `build_debug_test` wrappers should be used.
#[macro_export]
macro_rules! build_test_by_mode {
    ($in_debug_mode:expr, $source:expr) => {{
        $crate::helpers::Test::new($source, $in_debug_mode)
    }};
    ($in_debug_mode:expr, $source:expr, $stack_inputs:expr) => {{
        let inputs = $crate::helpers::ProgramInputs::new($stack_inputs, &[], vec![]).unwrap();

        $crate::helpers::Test {
            source: String::from($source),
            inputs,
            in_debug_mode: $in_debug_mode,
        }
    }};
    ($in_debug_mode:expr, $source:expr, $stack_inputs:expr, $advice_tape:expr, $advice_sets:expr) => {{
        let inputs =
            $crate::helpers::ProgramInputs::new($stack_inputs, $advice_tape, $advice_sets).unwrap();

        $crate::helpers::Test {
            source: String::from($source),
            inputs,
            in_debug_mode: $in_debug_mode,
        }
    }};
}
```